internal/controlplane/handlers_evalstatus.go

```diff
@@ -348,13 +348,27 @@ func buildProfileStatus(
 	}
 }
 
-// buildEvalResultAlertFrom
-func buildEvalResultAlertFrom(eval *db.ListRuleEvaluationsByProfileIdRow) *minderv1.EvalResultAlert {
-	return &minderv1.EvalResultAlert{
+// buildEvalResultAlertFromLRERow build the evaluation result alert from a
+// database row.
+func buildEvalResultAlertFromLRERow(eval *db.ListRuleEvaluationsByProfileIdRow) *minderv1.EvalResultAlert {
```

Comment:
For the alert part let's add a check similar to:
```go
	// If the alert is on and its metadata is valid, parse it and set the URL
	if dbRuleEvalStat.AlertMetadata.Valid && st.Alert.Status == string(db.AlertStatusTypesOn) {
		alertURL, err := getAlertURLFromMetadata(
			dbRuleEvalStat.AlertMetadata.RawMessage,
			fmt.Sprintf("%s/%s", st.EntityInfo["repo_owner"], st.EntityInfo["repo_name"]),
		)
		if err != nil {
			l.Err(err).Msg("error getting alert URL from metadata")
		} else {
			st.Alert.Url = alertURL
		}
	}
	return st, nil
```

This way we ensure we populate the alert URL only if there's a valid metadata and opened alert. Otherwise we leave the Url value empty(default) without error-ing.

Reply:
Cool, I added the check, PTAL @rdimitrov  :)